### ISSUE TITLE
last-modified timestamp for datastores

### ISSUE THREAD

**Issue by gotthardp** [sysrepo/sysrepo#2858]
I wonder if you ever considered adding a way to retrieve a timestamp indicating when a datastore was last modified. For the startup it could(?) be implemented simply by retrieving `fstat` of the corresponding `.../repository/data/*.startup` file, for the running it could be based on the `/dev/shm/sr_*.running`.

First, the RFC8040 (RESTCONF) says: *The server SHOULD maintain a last-modified timestamp for the datastore resource, defined in [Section 3.4](https://datatracker.ietf.org/doc/html/rfc8040#section-3.4).  This timestamp is only affected by configuration child data resources and MUST NOT be updated for changes to non-configuration child data resources. Last-modified timestamps for data resources are discussed in [Section 3.5](https://datatracker.ietf.org/doc/html/rfc8040#section-3.5).*

Second, an admin troubleshooting a sysrepo-based system may want to know whether the configuration has changed recently or not. (It may be recorded in a syslog, but the timestamp displayed somewhere in a system dashboard could be a nice initial indicator.)

What is your opinion on this subject? Could a timestamp like this be introduced into a sysrepo, or should we implement a custom plug-in for this purpose? I think that others, e.g. RESTCONF server implementers, may benefit from this feature too.

**Comment by michalvasko** [comment]
Yes, this seems like a nice feature I have nothing against even though there is the minor inconvenience of having to change DS plugin API by adding a new callback that is supposed to return the last modification timestamp. Then, I would expect these timestamps to be available in `sysrepo-monitoring` data, so that YANG needs to be changed and these new data populated.

That all should not be too difficult to implement but the amount of work is substantial and I currently have other more important work to do so it would be ideal if you implemented this yourself. Otherwise, I will at some point but no idea when.
